src/Analysers/AttributeAnnotationFactory.php

```diff
@@ -53,7 +53,7 @@ public function build(\Reflector $reflector, Context $context): array
                 // also look at parameter attributes
                 foreach ($reflector->getParameters() as $rp) {
                     foreach ([OAT\Property::class, OAT\Parameter::class, OAT\PathParameter::class] as $attributeName) {
-                        foreach ($rp->getAttributes($attributeName) as $attribute) {
+                        foreach ($rp->getAttributes($attributeName, \ReflectionAttribute::IS_INSTANCEOF) as $attribute) {
```

Comment:
Good find! I didn't know about the flag yet.

In fact, with the flag the `PathParameter::class` element can be removed from the foreach I suppose?

Reply:
I've been thinking about this before - hence the `PathParameter`  annotation.

I would argue that path and query are the most common cases and therefore we limit it to those two.

Reply:
> In fact, with the flag the PathParameter::class element can be removed from the foreach I suppose?

Yes, it's true :)

Reply:
@DerManoMann only one thing, for `Parameter` you have the `ParameterTrait`, but for `RequestBody` you don't.

In fact, for each base class, you need to create a trait, then it will be very easy to inherit any annotation/attribute class.

Reply:
Not sure I can follow you here...

Reply:
Ah, maybe I am. Yes, that would be handy but also quite some overhead for a non-quantifiable benefit.

Reply:
@DerManoMann About this feature.
As I said, If you want, I can try to introduce new Attributes.
In my project, I already have `QueryParameter` and `RequestBody`, 
Similar to `QueryParameter`, I can add `CookieParameter` and `HeaderParameter` attributes (without annotation class for them, because I need to extend the `Parameter` attribute class)

Reply:
> In fact, with the flag the PathParameter::class element can be removed from the foreach I suppose?

I won't do this, because this is a BC, I can just `@deprecate` it.

Reply:
Ah, I see. Maybe it could be like this then to avoid special treatment for `PathParameter`

```
                    foreach ([OA\Property::class, OA\Parameter::class] as $attributeName) {
                        foreach ($rp->getAttributes($attributeName, \ReflectionAttribute::IS_INSTANCEOF) as $attribute) {
```

Reply:
@ruscon Let me know if the above works for you...

Reply:
@DerManoMann Do you mean, removing `PathParameter`?
Yes, it will work, but if somebody already uses it, you will create a breaking change.

Reply:
No, I mean 
```
                    foreach ([OA\Property::class, OA\Parameter::class] as $attributeName) {
                        foreach ($rp->getAttributes($attributeName, \ReflectionAttribute::IS_INSTANCEOF) as $attribute) {
```

Reply:
Note that the alias changed from **OAT** to **OA**, so any attributes extended from the **annotation** base classes will match.

Reply:
> Note that the alias changed from OAT to OA, so any attributes extended from the annotation base classes will match.

@DerManoMann Yep, it works with `OA`.

Some tests crashed.

```bash
1) OpenApi\Tests\Annotations\AttributesSyncTest::testPropertyCompleteness with data set "HeaderParameter" ('OpenApi\Attributes\HeaderParameter')
ReflectionException: Class "OpenApi\Annotations\HeaderParameter" does not exist
```

should I add an Annotation version for the new Attributes?